src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -2704,6 +2704,8 @@ kernel_arg_transaction_execute (RpmostreedTransaction *transaction,
   OstreeSysroot *sysroot = rpmostreed_transaction_get_sysroot (transaction);
   int upgrader_flags = 0;
   auto command_line = static_cast<const char *>(vardict_lookup_ptr (self->options, "initiating-command-line", "&s"));
+  auto append_if_missing = static_cast<char **>(vardict_lookup_strv_canonical (self->options, "append-if-missing"));
+  auto delete_if_present = static_cast<char **>(vardict_lookup_strv_canonical (self->options, "delete-if-present"));
```

Comment:
I think for these we have to use `g_autofree char **append_if_missing` instead of `auto`. Otherwise, we'll leak these arrays. It's fine for the `vardict_lookup_ptr` version because it's referencing data inside the GVariant. But for arrays, it needs to create the array container itself and the `auto` keyword won't know that it needs to be freed. (Yeah, this is confusing. The TL;DR there is that we're in the process of moving C code to Rust, but using C++ in the transition to make that easier).

Reply:
changed and tested again, just in case :)